src/libostree/ostree-repo-pull.c

```diff
@@ -4935,13 +4938,13 @@ find_remotes_cb (GObject      *obj,
    * to be %NULL-safe. */
   g_ptr_array_set_free_func (results, (GDestroyNotify) repo_finder_result_free0);
 
-  /* FIXME: Add support for options:
-   *  - override-commit-ids (allow downgrades)
-   *
-   * Use case: multiple pulls of separate subdirs; want them to use the same
-   * configuration.
-   * Use case: downgrading a flatpak app.
-   */
+  if (data->options)
+    {
+      (void) g_variant_lookup (data->options, "override-commit-ids", "^a&s", &override_commit_ids);
+    }
+
+  if (override_commit_ids)
+    g_assert (g_strv_length ((gchar **) refs) == g_strv_length (override_commit_ids));
```

Comment:
Hmm, technically it would be more helpful if this precondition check was in the `ostree_repo_find_remotes_async()` function itself, so a failure would be reported to the caller immediately without leaving OSTree in a slightly less well-defined state.

You should probably also phrase it as an implication, i.e. `override_commit_ids != NULL ⇒ lengths are equal`:
```
g_return_if_fail (override_commit_ids == NULL || g_strv_length (…) == g_strv_length (…));
```

Reply:
Yeah but I guess we would have to add a `GError` parameter to `find_remotes_async` to give feedback to the caller.